src/cargo/core/profiles.rs

```diff
@@ -370,13 +268,170 @@ impl Profiles {
         }
         Ok(())
     }
+}
+
+/// Returns the hard-coded directory names for built-in profiles.
+fn predefined_dir_names() -> HashMap<InternedString, InternedString> {
+    [
+        (InternedString::new("dev"), InternedString::new("debug")),
+        (InternedString::new("test"), InternedString::new("debug")),
+        (InternedString::new("bench"), InternedString::new("release")),
+    ]
+    .into()
+}
+
+/// Returns the "root" profiles (dev/release)
+fn root_profiles(
+    profiles: &BTreeMap<InternedString, TomlProfile>,
+) -> Vec<(InternedString, ProfileMaker)> {
+    vec![
```

Comment:
We dont need a `Vec` here I guess? an `[(InternedString, ProfileMaker); 2]` should be sufficient.